File: frontend/src/types/issue.ts
export type IssueStatus = "TODO" | "IN_PROGRESS" | "REVIEW" | "DONE";
export type IssuePriority = "LOW" | "MEDIUM" | "HIGH";

export interface Issue {
  id: number;
  projectId: number;
  projectKey: string;
  projectName: string;
  title: string;
  description: string | null;
  status: IssueStatus;
  statusLabel: string;
  priority: IssuePriority;
  priorityLabel: string;
  assigneeName: string | null;
  dueDate: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface IssueRequest {
  projectId: number;
  title: string;
  description: string;
  status: IssueStatus;
  priority: IssuePriority;
  assigneeName: string;
  dueDate: string | null;
}

File: backend/src/main/java/com/projectmanagementapp/domain/model/Issue.java
package com.projectmanagementapp.domain.model;

import java.time.LocalDate;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Issue {
    private Long id;
    private Long projectId;
    private String projectKey;
    private String projectName;
    private String title;
    private String description;
    private IssueStatus status;
    private IssuePriority priority;
    private String assigneeName;
    private LocalDate dueDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
